src/background_workers.c

```diff
@@ -188,11 +189,29 @@ static RedisAI_RunInfo **_BGThread_BatchOperations(RunQueueInfo *run_queue_info,
         return batch_rinfo;
     }
 
+    // Set the batch to be ready by default (optimistic), change it during run.
+    *batchReady = true;
+    bool timeout = false;
+    // If minbatchsize has been set and we are not past it, we check
+    // if the timeout for min batch has expired, in which case we proceed
+    // anyway
+    if (minbatchsize > 0 && minbatchtimeout > 0) {
```

Comment:
Maybe we can move the timeout checking to the beginning of the loop (line 214) and avoid the duplication in lines 290-278

Reply:
if the queue is empty, you need to see that the pending task is not timedout